### ISSUE TITLE
empty record for Gift card file in system_files table

### ISSUE THREAD

**Issue by dathwa** [OFFLINE-GmbH/oc-mall-plugin#593]
Hi.
Something curious has started happenning with gift cards.
I've added a gift card option, and used code here: https://offline-gmbh.github.io/oc-mall-plugin/digging-deeper/virtual-products.html#generate-user-specific-product-files

This was working well, a few months back. After returning to the project this week, it's no longer working.
The file is generated OK. We have the path, but after adding and saving, a record is created in system_files, but disk_name, file_name, file_size and content_type are empty.
```
$grant->file = $path;
// Don't forget to save the changes!
$grant->save();
```
Is this definitely still working? Has something maybe changed which is breaking the attachment of the file to the model?
If you could do just one tiny wee test on your system to confirm all is OK I would be so grateful.

Thank-you.

**Comment by tobias-kuendig** [comment]
What version of October are you on? IIRC there were some changes made to the File model that could have caused this issue. Could you try and see if these changes could be the culprit:

https://github.com/octobercms/library/commit/6cd6af6940ca15598ad1bab64e2db4a005d9d38f

**Comment by dathwa** [comment]
Current build is 469 and I don't see these changes in `src/Database/Attach/File.php `

**Comment by dathwa** [comment]
Hi T. Quick update. Testing on windows.
Just installed latest xampp to support php 7.4. Installed a fresh ocms (469) with mall and mall theme.
installed this: [https://gist.github.com/tobias-kuendig/49ec99fc080fc6ba3824024f965eeafc]
modified line 27 for the seeded gift category: `return $category->code === 'gift-cards';`
The file is generated ok in the path /app/gift_card_2.txt
offline_mall_product_file_grants record created OK.
But see this: https://prnt.sc/uutyux
No disk_name for ANY of the seed data or the gift card.
debug is on, nothing in logs.

Btw, our apache/linux sever is also having issues as reported in initial post.

Any ideas, anyone?

**Comment by tobias-kuendig** [comment]
This might be related to #599.

**Comment by dathwa** [comment]
Hi Tobias. This issue was indeed related to [octobercms/library@6cd6af6](https://github.com/octobercms/library/commit/6cd6af6940ca15598ad1bab64e2db4a005d9d38f).
Thank-you for your help.
My current build is 469 and line 37 in `src/Database/Attach/File.php` was `protected $guarded = ['disk_name'];`.
i changed to `protected $guarded = [];` and this fixed it.
But I don't understand why this has not been updated automatically?
Even on a new ocms installation, using the installer, this file is not updated. Is this an edge issue?

**Comment by tobias-kuendig** [comment]
I'm not sure about the state of this issue on October's side. I know it is fixed in October 1.1. The fix should have been backported to 1.0 as well, not sure why you did not get it:

https://github.com/octobercms/library/pull/521/files
